components/core/Tutorial/Header.tsx

```diff
@@ -0,0 +1,100 @@
+import Image from 'next/image';
+import { useRouter } from 'next/router';
+import React, { useEffect, useState } from 'react';
+import tutorials from 'pages/tutorials/developing/_meta.en-US.json';
+import { Tutorial } from 'types/tutorial';
+
+export function getTutorialByRouteName(
+  json: Record<string, Tutorial>,
+  routeName: string,
+): Tutorial | null {
+  const hashTable: Record<string, Tutorial> = {};
+
+  // Create hash table based on route/href name - It is a faster than loop implementation
+  for (let key in json) {
+    const item = json[key];
+    if (item && item.href) {
+      hashTable[item.href] = item;
+    }
+  }
+
+  return hashTable[routeName] || null;
+}
+
+export default function Header() {
+  const router = useRouter();
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [tutorial, setTutorial] = useState<Tutorial | null>(null);
+
+  const ogImage = `http://localhost:3000/api/og?title=${tutorial?.title}&category=${tutorial?.category}`;
```

Comment:
We'll need to change this to a relative path:
```suggestion
  const ogImage = `/api/og?title=${tutorial?.title}&category=${tutorial?.category}`;
```

Reply:
Missed that 🤦

Sure, thanks Chase!